client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditForm.tsx

```diff
@@ -10,13 +10,14 @@ import {
 import {
   StockItemSearchInput,
   ItemRowFragment,
+  ItemStockOnHandFragment,
 } from '@openmsupply-client/system';
 import { useStocktake } from '../../../api';
 
 interface StocktakeLineEditProps {
   item: ItemRowFragment | null;
   mode: ModalMode | null;
-  onChangeItem: (item: ItemRowFragment | null) => void;
+  onChangeItem: (item: ItemStockOnHandFragment | null) => void;
```

Comment:
I've changed to use a shared type - the different item api calls returned different objects. Am now using a fragment that represents the common intersection of properties